### PR TITLE
Add table update/patch functionality

### PR DESCRIPTION
**Pull request by jskeet**
This is a copy/paste of dataset functionality, with the only
difference being that specifying a table reference requires three
pieces of information instead of two.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryTable.cs [hunk 1]
```diff
@@ -197,14 +197,53 @@ internal BigQueryTable(BigQueryClient client, Table resource)
...
+        {
+            if (matchEtag)
+            {
+                resource.ETag = Resource.ETag;
+            }
+            return _client.PatchTable(Reference, resource, options);
+        }
+        
+        /// <summary>
+                 /// Asynchronously uploads a stream of CSV data to this table.
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryTable.cs]: Indent is wrong from this line to the end of the xmldoc.

Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryTable.cs]: Gosh - I wonder how that happened. Thanks!

**Comment by NPSF3000** [comment]
Hi @jskeet, it's nice to see someone of your reputation when chasing down bugs ;)  

I just read through the code and saw the "client.PatchTable(datasetId, tableId, new Table { FriendlyName = "Patched table" });" snippet, but unfortunately this is not yet in nuget.

What's the process between testing and release?  Should I just grab the repo myself and build?

**Comment by jskeet** [comment]
@NPSF3000: We do a release periodically, typically after we've added a whole set of changes - often if we're not expecting more changes imminently. I'm in the middle of doing some more work, so there may not be another release for a week or so. If you really need this *right now*, you can pull the source, but typically I'd recommend waiting until there's a release.

You can always get the `BigqueryService` from the client and call `Patch` yourself, of course - this library just makes things a bit simpler.

**Comment by NPSF3000** [comment]
@jskeet thanks for the info.  Doing some once off maintenance now so I'll figure it out.  Glad to see .Net getting some love!

**Pull request merged**